core-bundle/src/Resources/contao/classes/Backend.php

```diff
@@ -273,6 +275,21 @@ public static function handleRunOnce()
 
 			System::log("File $strRelpath ran once and has then been removed successfully", __METHOD__, TL_GENERAL);
 		}
+
+		$appDir = System::getContainer()->getParameter('kernel.project_dir') . '/app';
+
+		if (!is_dir($appDir))
+		{
+			return;
+		}
+
+		$finder = Finder::create()->files()->in($appDir);
+
+		// Remove the app folder if there are no more files in it
+		if (!$finder->hasResults())
+		{
+			(new Filesystem())->remove($appDir);
```

Comment:
We might want to catch (and ignore) and `IOExceptions` happening here, e.g. if the directory could not be removed (due to access permissions or whatever, we don't really care).

Reply:
Do we do the same in the 'runonces' itself?

Reply:
Changed in ca21d34571df4cacc223b6bcef79886c2c82d5b7.

Reply:
> Do we do the same for the 'runonces' itself? Edit: no, looks like we are failing there with an exception.

Yes, because we don't want to run the file over and over again.